src/sentry/integrations/client.py

```diff
@@ -236,3 +237,19 @@ def _request(self, method, path, **kwargs):
         self.auth.refresh_token()
         kwargs = self.bind_auth(**kwargs)
         return ApiClient._request(self, method, path, **kwargs)
+
+
+class OAuth2ApiClient(ApiClient):
+    def __init__(self, identity, *args, **kwargs):
+        super(OAuth2ApiClient, self).__init__(*args, **kwargs)
+        self.identity = identity
+
+    def check_auth(self):
```

Comment:
This is small, but I sort of wish I had a different name for this. Open to suggestions.

Reply:
The name sounds good to me actually